client/driver/executor/executor_linux.go

```diff
@@ -221,181 +365,240 @@ func (e *UniversalExecutor) Stats() (*cstructs.TaskResourceUsage, error) {
-	e.cmd.SysProcAttr.Credential.Gid = uint32(gid)
-	e.cmd.SysProcAttr.Credential.Groups = gids
 
-	e.logger.Printf("[DEBUG] executor: running as user:group %d:%d with group membership in %v", uid, gid, gids)
+	cfg.ReadonlyPaths = []string{
+		"/proc/sys", "/proc/sysrq-trigger", "/proc/irq", "/proc/bus",
+	}
 
-	return nil
+	cfg.Devices = lconfigs.DefaultAutoCreatedDevices
+	cfg.Mounts = []*lconfigs.Mount{
+		{
+			Source:      "tmpfs",
+			Destination: "/dev",
+			Device:      "tmpfs",
+			Flags:       syscall.MS_NOSUID | syscall.MS_STRICTATIME,
+			Data:        "mode=755",
+		},
+		{
+			Source:      "devpts",
+			Destination: "/dev/pts",
+			Device:      "devpts",
+			Flags:       syscall.MS_NOSUID | syscall.MS_NOEXEC,
+			Data:        "newinstance,ptmxmode=0666,mode=0620,gid=5",
+		},
+		{
+			Device:      "tmpfs",
+			Source:      "shm",
+			Destination: "/dev/shm",
+			Data:        "mode=1777,size=65536k",
+			Flags:       defaultMountFlags,
+		},
+		{
+			Source:      "mqueue",
+			Destination: "/dev/mqueue",
+			Device:      "mqueue",
+			Flags:       defaultMountFlags,
+		},
+		{
+			Source:      "sysfs",
+			Destination: "/sys",
+			Device:      "sysfs",
+			Flags:       defaultMountFlags | syscall.MS_RDONLY,
+		},
+	}
 }
 
-// configureChroot configures a chroot
-func (e *UniversalExecutor) configureChroot() error {
-	if e.cmd.SysProcAttr == nil {
-		e.cmd.SysProcAttr = &syscall.SysProcAttr{}
+func configureCgroups(cfg *lconfigs.Config, command *ExecCommand) error {
+
+	// If resources are not limited then manually create cgroups needed
+	if !command.ResourceLimits {
+		return configureBasicCgroups(cfg)
 	}
-	e.cmd.SysProcAttr.Chroot = e.ctx.TaskDir
-	e.cmd.Dir = "/"
 
-	e.fsIsolationEnforced = true
-	return nil
-}
+	id := uuid.Generate()
+	cfg.Cgroups.Path = filepath.Join(defaultCgroupParent, id)
+	if command.Resources.MemoryMB > 0 {
+		// Total amount of memory allowed to consume
+		cfg.Cgroups.Resources.Memory = int64(command.Resources.MemoryMB * 1024 * 1024)
+		// Disable swap to avoid issues on the machine
+		var memSwappiness uint64 = 0
+		cfg.Cgroups.Resources.MemorySwappiness = &memSwappiness
+	}
 
-// getAllPids returns the pids of all the processes spun up by the executor. We
-// use the libcontainer apis to get the pids when the user is using cgroup
-// isolation and we scan the entire process table if the user is not using any
-// isolation
-func (e *UniversalExecutor) getAllPids() (map[int]*nomadPid, error) {
-	if e.command.ResourceLimits || e.command.BasicProcessCgroup {
-		manager := getCgroupManager(e.resConCtx.groups, e.resConCtx.cgPaths)
-		pids, err := manager.GetAllPids()
-		if err != nil {
-			return nil, err
-		}
-		np := make(map[int]*nomadPid, len(pids))
-		for _, pid := range pids {
-			np[pid] = &nomadPid{
-				pid:           pid,
-				cpuStatsTotal: stats.NewCpuStats(),
-				cpuStatsSys:   stats.NewCpuStats(),
-				cpuStatsUser:  stats.NewCpuStats(),
-			}
+	if command.Resources.CPU < 2 {
```

Comment:
I know you didn't touch this code, but I wonder what the author had against 1?! AFAICT this has been 2 from its inception.